devtools/gradle/gradle-model/src/main/java/io/quarkus/gradle/dependency/ApplicationDeploymentClasspathBuilder.java

```diff
@@ -70,10 +70,12 @@ public static void initConfigurations(Project project) {
                         configContainer.getByName(JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME));
 
         // enable the Panache annotation processor on the classpath, if it's found among the dependencies
-        configContainer.getByName(JavaPlugin.ANNOTATION_PROCESSOR_CONFIGURATION_NAME).getIncoming()
-                .beforeResolve(annotationProcessors -> {
-                    Set<ResolvedArtifact> compileClasspathArtifacts = configContainer
-                            .getByName(JavaPlugin.COMPILE_CLASSPATH_CONFIGURATION_NAME).getResolvedConfiguration()
+        configContainer.getByName(JavaPlugin.ANNOTATION_PROCESSOR_CONFIGURATION_NAME)
+                .defaultDependencies(annotationProcessors -> {
```

Comment:
Could you please explain the reason for changing to the `defaultDependencies(...)` here? The javadoc says
````
Execute the given action if the configuration has no defined dependencies when it first participates in dependency resolution.
````
Would it still work if there already are some default dependencies and the `quarkus-panache-common` is not among them?

Reply:
I followed the error message but as you mentioned this is not what we are looking for. 
The `withDependencies` should be ok. I will give it a try.

Reply:
I replace the `defaultDependencies` by the `withDependencies` which always runs before dependency resolution.